controllers/storagecluster/cephcluster.go

```diff
@@ -1707,3 +1718,52 @@ func getFullRatio(specifiedRatio *float64) *float64 {
 	defaultRatio := 0.85
 	return &defaultRatio
 }
+
+// updateRookCephOperatorHostNetwork updates the rook-ceph-operator CSV to enable host network
+func updateRookCephOperatorHostNetwork(ctx context.Context, c client.Client, opNamespace string) error {
+	csvList := &opv1a1.ClusterServiceVersionList{}
+	labelSelector := labels.SelectorFromSet(map[string]string{
+		fmt.Sprintf("operators.coreos.com/rook-ceph-operator.%v", opNamespace): "",
+	})
+	listOpts := &client.ListOptions{
+		Namespace:     opNamespace,
+		LabelSelector: labelSelector,
+	}
+
+	// List the ClusterServiceVersion for rook-ceph-operator
+	err := c.List(ctx, csvList, listOpts)
+	if err != nil {
+		return err
+	}
+
+	// Check if we found exactly one ClusterServiceVersion for rook-ceph-operator
+	if len(csvList.Items) == 0 {
+		return fmt.Errorf("no ClusterServiceVersion found for rook-ceph-operator in namespace %s", opNamespace)
+	}
```

Comment:
This is wrong, you need to check if the CSV is the current CSV. During upgrades, even stale upgrades you will always find more then one csv